### PR TITLE
Resource terminate after & compact option to generate

### PR DESCRIPTION
**Pull request by anniezhang77**
risk: low

### CODE REVIEW

File: spade-core/src/main/scala/com/salesforce/mce/spade/workflow/Resource.scala [hunk 1]
```diff
@@ -14,5 +14,6 @@ case class Resource[+R](
   name: String,
   resourceType: String,
   resourceSpec: Json,
-  maxAttempt: Int
+  maxAttempt: Int,
+  terminateAfter: Option[Double]
```
Comment (realstraw) [inline spade-core/src/main/scala/com/salesforce/mce/spade/workflow/Resource.scala]: let's make this `Duration` and convert to `Double` when converting `workflow.Resource` to `WorkflowRequest.Resource`

File: version.sbt [hunk 2]
```diff
@@ -1 +1 @@
- ThisBuild / version := "0.8.1"
+ ThisBuild / version := "0.8.2"
```
Comment (realstraw) [inline version.sbt]: let's also do `0.9.0` here

File: spade-core/src/main/scala/com/salesforce/mce/spade/workflow/Resource.scala (hunk 1 above)
Comment (anniezhang77) [reply spade-core/src/main/scala/com/salesforce/mce/spade/workflow/Resource.scala]: done

File: spade-core/src/main/scala/com/salesforce/mce/spade/orchard/WorkflowRequest.scala [hunk 3]
```diff
@@ -73,7 +74,8 @@ object WorkflowRequest {
           r.name,
           r.resourceType,
           r.resourceSpec,
-          r.maxAttempt
+          r.maxAttempt,
+          r.terminateAfter.map(_.toHours.toDouble)
```
Comment (realstraw) [inline spade-core/src/main/scala/com/salesforce/mce/spade/orchard/WorkflowRequest.scala]: let's make it have minute level precision instead. currently 1.5 hours would be converted to 1 hour.

Comment (anniezhang77) [reply spade-core/src/main/scala/com/salesforce/mce/spade/orchard/WorkflowRequest.scala]: done

**Review by realstraw — APPROVED** [review]

**Pull request merged**